File: sdk-example/python/user/jumpbox_service_post_users.py
import time
import os
import ibm_gdsc_sdk_software
from ibm_gdsc_sdk_software.models.jumpboxv3_post_users_bulk_request import Jumpboxv3PostUsersBulkRequest
from ibm_gdsc_sdk_software.models.jumpboxv3_post_users_bulk_response import Jumpboxv3PostUsersBulkResponse
from ibm_gdsc_sdk_software.rest import ApiException
from pprint import pprint
from ibm_gdsc_sdk_software.models.jumpboxv3_user import Jumpboxv3User
import json
# Defining the host is optional and defaults to http://localhost
# See configuration.py for a list of all supported configuration parameters.
# configuration = ibm_gdsc_sdk_software.Configuration(
#     host = ""
# )

# The client must configure the authentication and authorization parameters
# in accordance with the API server security policy.
# Examples for each auth method are provided below, use the example that
# satisfies your auth use case.
# APIKEY=os.environ['APIKEY']
# SECRET=os.environ['SECRET']
# Configure HTTP basic authorization: BasicAuth
configuration = ibm_gdsc_sdk_software.Configuration(
    host = os.environ['URL'],
    username = os.environ['API_USERNAME'],
    password = os.environ['API_PASSWORD']
)

# Configure API key authorization: ApiKeyAuth
#configuration.api_key['ApiKeyAuth'] = os.environ['API_KEY']
configuration.verify_ssl = False
# Uncomment below to setup prefix (e.g. Bearer) for API key, if needed
#configuration.api_key_prefix['ApiKeyAuth'] = 'Basic'
# Enter a context with an instance of the API client

with ibm_gdsc_sdk_software.ApiClient(configuration) as api_client:
    # Create an instance of the API class
    api_instance = ibm_gdsc_sdk_software.JumpboxServiceApi(api_client)
    # create an instance of Jumpboxv3User from a JSON string
    json = '{"users":[{"distinguished_name":"uid=2J9097897,c=us,ou=bluepages,o=ibm.com","roles":["admin","default"]}]}'
    # create an instance of Jumpboxv3PostUsersBulkRequest from a JSON string
    jumpboxv3_post_users_bulk_request_instance = Jumpboxv3PostUsersBulkRequest.from_json(json)
 
    try:
        # Summary: Post users Description: Create users.
        api_response = api_instance.jumpbox_service_post_users(jumpboxv3_post_users_bulk_request_instance)
        print("The response of JumpboxServiceApi->jumpbox_service_post_users:\n")
        pprint(api_response)
    except Exception as e:
        print("Exception when calling JumpboxServiceApi->jumpbox_service_post_users: %s\n" % e)

File: sdk-example/typescript/src/user/jumpboxServicePostUsers.ts
import { AuthMethodsConfiguration, ServerConfiguration, Middleware, RequestContext, ResponseContext, createConfiguration, JumpboxServiceApi, Jumpboxv3PostUsersBulkRequest, Jumpboxv3User } from '@ibm-gdsc/ibm-gdsc-sdk-software';

// Covers all auth methods included in your OpenAPI yaml definition
// Go to the Guardium Security Center website and create an API key , copy the one start with "Basic" and then export here:
// export API_KEY="YOUR_API_KEY"
// Set the URL of your server as environment variable
//export URL=https://test.ibm.com:8443

const authConfig: AuthMethodsConfiguration = {
    'ApiKeyAuth': process.env.API_KEY
}

const url = process.env.URL ?? 'https://localhost:8443'
const variableConfiguration = {}
const serverConfig = new ServerConfiguration(url, variableConfiguration)
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'

// Implements a simple middleware to modify requests before (`pre`) they are sent
// and after (`post`) they have been received 
class Test implements Middleware {
    pre(context: RequestContext): Promise<RequestContext> {
        // Modify context here and return
        return Promise.resolve(context);
    }

    post(context: ResponseContext): Promise<ResponseContext> {
        return Promise.resolve(context);
    }

}

// Create configuration parameter object
const configurationParameters = {
    httpApi: undefined, // Can also be ignored - default is usually fine
    baseServer: serverConfig, // First server is default
    authMethods: authConfig, // No auth is default
    promiseMiddleware: [new Test()],
}

// Convert to actual configuration
const config = createConfiguration(configurationParameters);

// Use configuration
const apiInstance = new JumpboxServiceApi(config);
 
const user: Jumpboxv3User = {
    distinguishedName: 'uid=2J9097897,c=us,ou=bluepages,o=ibm.com',
    roles: ['admin', 'default'],
}

const request:Jumpboxv3PostUsersBulkRequest = {
    users: [ user], 
} 

const postUsers = async (request: Jumpboxv3PostUsersBulkRequest) => {
    const response = await apiInstance.jumpboxServicePostUsers(request);
    console.log('API called successfully. Returned data: ' + JSON.stringify(response));
  
}

postUsers(request);